Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -207,40 +226,52 @@ private void StopShooting(EntityUid uid, GunComponent gun)
     /// <summary>
     /// Attempts to shoot at the target coordinates. Resets the shot counter after every shot.
     /// </summary>
-    public void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun, EntityCoordinates toCoordinates)
+    public void AttemptShoot(EntityUid? user, EntityUid gunUid, GunComponent gun, EntityCoordinates? toCoordinates = null)
     {
-        gun.ShootCoordinates = toCoordinates;
+        if (toCoordinates == null)
+            gun.ShootCoordinates = new EntityCoordinates(gunUid, new Vector2(0, -1)); //Forward vector
+        else
+            gun.ShootCoordinates = toCoordinates;
+
         AttemptShoot(user, gunUid, gun);
         gun.ShotCounter = 0;
     }
```

Comment:
Adding a bunch of null coalescing to methods makes it hard to discern what's happening (in this case someone might not realise they forgot to pass an arg in), just have the caller handle it or worst case make a new method that doesn't take in entitycoordinates that calls this.